server/src/jira/jira.service.ts

```diff
@@ -0,0 +1,38 @@
+import { Inject, Injectable, Scope } from "@nestjs/common";
+import { AxiosResponse } from "axios";
+import { AxiosService } from "src/axios/axios.service";
+import { ConfigService } from "src/config/config.service";
+import { JiraTokens } from "./jira.types";
+import { REQUEST } from "@nestjs/core";
+import { Request } from "express";
+
+@Injectable({ scope: Scope.REQUEST })
+export class JiraService {
+  constructor(
+    private axiosService: AxiosService,
+    private configService: ConfigService,
+    @Inject(REQUEST) private request: Request,
+  ) {}
+
+  async getFreshTokens(refreshToken: string): Promise<JiraTokens> {
+    const { clientId, clientSecret, tokenUrl } = this.configService.config.jira;
+    const res: AxiosResponse<{ refresh_token: string; access_token: string }> =
```

Comment:
I would write this using the `post` method's type parameter:

```
const res = await this.axiosService.post<{ refresh_token: string; access_token: string }>( ... )
```

This way there's no need to import `AxiosResponse` type and you get better type safety as TS won't attempt to _coerce_ the return type to your variable but rather _assigns_ it. (For example, if `AxiosService.post` was modified to return `any`, the variable `res` would, with the current code, still have the same `AxiosResponse` type, hiding possible typing problems downstream.)